python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+"""SignalExchangeAPI implementation for Tech Against Terrorism Hash List API"""
+
+
+from functools import lru_cache
+import json
+import tempfile
+import requests
+import time
+import typing as t
+from dataclasses import dataclass, field
+
+from threatexchange.exchanges.clients.techagainstterrorism import api
+
+from threatexchange.exchanges import fetch_state as state
+from threatexchange.exchanges import signal_exchange_api
+from threatexchange.exchanges import auth
+from threatexchange.exchanges.collab_config import (
+    CollaborationConfigBase,
+    CollaborationConfigWithDefaults,
+)
+from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.signal_type.pdq.signal import PdqSignal
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.signal_type.url import URLSignal
+from threatexchange.signal_type.raw_text import RawTextSignal
+
+
+@dataclass
+class TATCheckpoint(state.FetchCheckpointBase):
+    """
+    Tech Against Terrorism Hash List revolves around fetching
+    a JSON file which contains a list of hashes. The pre-signed URL
+    for the JSON file is returned from the API response.
+
+    The Hash List is updated nightly and new hashes are appended to the list.
+
+    For more information on our collection process please visit: https://terrorismanalytics.org/about/how-it-works
+    For our Hash List documentation: https://terrorismanalytics.org/docs/hash-list-v1
+    """
+
+    last_fetch_time: int
+
+    def is_stale(self) -> bool:
+        """
+        Consider stale after 24 hours of not fetching
+        """
+        return time.time() - self.last_fetch_time > 3600 * 24
+
+    @classmethod
+    def from_tat_fetch(cls, response: api.TATHashListResponse) -> "TATCheckpoint":
+        return cls(int(response.created_on.timestamp()))
+
+
+@dataclass
+class TATSignalMetadata(state.FetchedSignalMetadata):
+    """Not sure what should live here"""
```

Comment:
This would be where tags would be if your API supports them. I'm guessing this a broadcast-only API from TAT -> consumers, with no feedback mechanisms? If so, there might not be anything here.

Reply:
That's right the hash list is only a broadcast API. Will `pass` be what we want here ? I know `SignalExchangeAPIWithSimpleUpdates` and `SignalExchangeAPI` both expect it to be passed.

Reply:
To follow up, we do have a hash verification service, but this may be a follow up piece of work as my PM specifically mentioned Hash List and this feature is currently time boxed.

I would be happy to return in the future and implement the hash verification.